### ISSUE TITLE
Can't create trend chart

### ISSUE THREAD

**Issue by pmatos** [alex-hhh/ActivityLog2#34]
Following my pursuits to using ActivityLog2 to determine my CP, W', etc I tried creating a trend chart of best avg as mentioned in `critical-power.md`. However I am getting:

```
instantiate: no argument for required init variable
  init variable name: trend-chart-class
  instantiated class name: trend-chart-pane%
  context...:
   /home/pmatos/installs/racket-6.12/collects/racket/private/class-internal.rkt:4610:0: obj-error35
   /home/pmatos/installs/ActivityLog2-rel-1.5.0/rkt/view-trends.rkt:186:2
   /home/pmatos/installs/racket-6.12/collects/racket/private/class-internal.rkt:3553:0: continue-make-object
   /home/pmatos/installs/racket-6.12/collects/racket/private/class-internal.rkt:3507:0: do-make-object
   /home/pmatos/installs/ActivityLog2-rel-1.5.0/rkt/view-trends.rkt:340:4: on-new-chart
   /home/pmatos/installs/racket-6.12/share/pkgs/gui-lib/mred/private/wx/common/queue.rkt:428:6
   /home/pmatos/installs/racket-6.12/share/pkgs/gui-lib/mred/private/wx/common/queue.rkt:479:32
   /home/pmatos/installs/racket-6.12/share/pkgs/gui-lib/mred/private/wx/common/queue.rkt:627:3
```

**Comment by pmatos** [comment]
I started to try to fix this in `wip-issue34` branch on my fork but the rabbit hole goes deeper than the missing arguments. It seems that I don't really know what the `restore-data` is supposed to be. I passed `(send tc get-restore-data)` but this returns `#f` which causes issues down the line.

**Comment by alex-hhh** [comment]
Well, this is embarrassing, I didn't create a test trend chart as part of a refactoring task.  

Try the patch below.  `restore-data` is not needed for new charts, and I simply omitted the `restore-from` in the trends chart initialization.  Each trends chart saves its parameters so they can be re-edited, and these parameters are in the `restore-data`.  The contents of the restore data is trends chart specific, so the rest of the code treats it as "almost" opaque, but it is a hash containing the parameters in the dialog box.

```diff
 rkt/trend-charts/view-trends.rkt | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/rkt/trend-charts/view-trends.rkt b/rkt/trend-charts/view-trends.rkt
index 5c90b19..1b6f961 100644
--- a/rkt/trend-charts/view-trends.rkt
+++ b/rkt/trend-charts/view-trends.rkt
@@ -184,7 +184,7 @@
 
 (define trend-chart-pane%
   (class panel%
-    (init-field parent info-tag trend-chart-class database restore-data)
+    (init-field parent info-tag trend-chart-class database [restore-data #f])
     (super-new [parent parent] [style '(deleted)])
 
     (define trend-chart #f)
@@ -194,7 +194,8 @@
     (define (maybe-initialize)
       (unless trend-chart
         (set! trend-chart (new trend-chart-class [database database]))
-        (send trend-chart restore-from restore-data)
+        (when restore-data
+          (send trend-chart restore-from restore-data))
         (set! graph-pb (new snip-canvas% [parent this]))))
 
     (define/public (get-name)
@@ -340,11 +341,12 @@
     (define (on-new-chart)
       (let ((ct (send (new new-trend-chart-dialog%) show-dialog parent)))
         (when ct
-          (let ((pane (let ((tc (new (tdecl-class ct) [database database])))
-                        (new trend-chart-pane%
-                             [parent trend-charts-panel]
-                             [info-tag (tdecl-tag ct)]
-                             [trend-chart tc]))))
+          (let ((pane (new trend-chart-pane%
+                           [parent trend-charts-panel]
+                           [info-tag (tdecl-tag ct)]
+                           [trend-chart-class (tdecl-class ct)]
+                           [database database]
+                           [restore-data #f])))
             (when (send pane interactive-setup parent)
               (set! trend-charts (append trend-charts (list pane)))
               (send trend-charts-panel append (send pane get-name))
```

**Comment by alex-hhh** [comment]
I think I am going to fix this on a `release-1.5-fixes` branch and release a 1.5.1 version....
